frontend/src/stores/DataStore.js

```diff
@@ -1,25 +1,297 @@
+        this.savedDataset.push(output);
+    }
+
+    @action saveAdverseDirection = (name, value, id) => {
+        this.savedDataset.map((item, i) => {
+            if (item.id == id) {
+                item[name] = value;
+            }
+        });
+    };
+
+    @action deleteDataset = id => {
+        var index = this.savedDataset.findIndex(item => item.id == id);
+
+        if (index > -1) {
+            this.savedDataset.splice(index, 1);
+        }
+    };
+
+    @action updateDataset(setId, name, value, id) {
+        this.savedDataset[setId][name][id] = value;
+    }
+
+    @action toggleDataset = idx => {
+        var obj = this.savedDataset.find(item => item.id == idx);
+        obj["enabled"] = !obj["enabled"];
+    };
+
+    @action deleteForm() {
+        this.inputForm.datasets = [];
+        this.inputForm.model_type = "";
+        this.inputForm.dataset_name = "";
+    }
+
+    @action toggleModal() {
+        this.modal = !this.modal;
+    }
+
+    @action addUsersInput = (name, value) => {
+        this.usersInput[name] = value;
+    };
+
+    @action closeDataForm() {
+        this.dataform = false;
+    }
+    @action setConfig = config => {
+        this.config = config;
+    };
+
+    @action toggleModelsCheckBox = (model, checked, value) => {
+        let models = this.getModelTypeList();
+
+        if (model.includes("bayesian_model_average") && checked) {
+            this.prior_weight_models.push(model);
+        } else if (model.includes("bayesian_model_average") && !checked) {
+            let index = this.prior_weight_models.indexOf(model);
+            this.prior_weight_models.splice(index, 1);
+        }
+
+        if (this.prior_weight_models.length) {
+            this.prior_weight = 100;
+            this.prior_weight = this.prior_weight / this.prior_weight_models.length;
+        }
+        models.map(item => {
+            item.values.map(val => {
+                if (val.name === model) {
+                    val.isChecked = !val.isChecked;
+                } else if (
+                    model.split("-")[1] == "All" &&
+                    model.split("-")[0] == val.name.split("-")[0] &&
+                    !val.isDisabled
+                ) {
+                    val.isChecked = !val.isChecked;
+                }
+
+                if (this.prior_weight_models.includes(val.name)) {
+                    val.prior_weight = this.prior_weight;
+                } else if (
+                    val.name.includes("bayesian_model_average") &&
+                    !this.prior_weight_models.includes(val.name)
+                ) {
+                    val.prior_weight = 0;
+                }
+            });
+        });
+    };
+
+    //returns the dataset which are enabled
+    @action getEnabledDataset() {
+        let obj = toJS(this.savedDataset).filter(item => item.enabled == true);
+        return obj;
+    }
+
+    //returns enabled model types
+    @action getModels() {
```

Comment:
should this also be computed? as i understand it, actions should only modify content in `observables`

Reply:
changed to @computed